sql/core/src/main/scala/org/apache/spark/sql/execution/SparkQl.scala

```diff
@@ -64,6 +85,53 @@ private[sql] class SparkQl(conf: ParserConf = SimpleParserConf()) extends Cataly
         val tableIdent = extractTableIdent(nameParts)
         RefreshTable(tableIdent)
 
+      case Token("TOK_CREATEDATABASE", Token(databaseName, Nil) :: createDatabaseArgs) =>
+        val Seq(
+          allowExisting,
+          dbLocation,
+          databaseComment,
+          dbprops) = getClauses(Seq(
+          "TOK_IFNOTEXISTS",
+          "TOK_DATABASELOCATION",
+          "TOK_DATABASECOMMENT",
+          "TOK_DATABASEPROPERTIES"), createDatabaseArgs)
+        val location = dbLocation.map {
+          case Token("TOK_DATABASELOCATION", Token(loc, Nil) :: Nil) => unquoteString(loc)
+        }
+        val comment = databaseComment.map {
+          case Token("TOK_DATABASECOMMENT", Token(comment, Nil) :: Nil) => unquoteString(comment)
+        }
+        val props: Map[String, String] = dbprops.toSeq.flatMap {
+          case Token("TOK_DATABASEPROPERTIES", propList) =>
+            propList.flatMap(extractProps(_, "TOK_DBPROPLIST", "TOK_TABLEPROPERTY"))
+        }.toMap
+        CreateDatabase(databaseName, allowExisting.isDefined, location, comment, props)(node.source)
+
+      case Token("TOK_CREATEFUNCTION", func :: as :: createFuncArgs) =>
```

Comment:
I am not sure this pattern works for functions with a dotted name. For example

    create function s.t as 'foo.bar.func'

Produces the following AST:

    TOK_CREATEFUNCTION 1, 0, 10, 16
    :- s 1, 4, 4, 16
    :- t 1, 6, 6, 18
    +- 'foo.bar.func' 1, 10, 10, 23

Notice how the name is splitted into two top-level ASTNodes.

I think we need to change the parser rule here.